pippy/IR.py

```diff
@@ -597,6 +597,14 @@ def _from_traced(mod : torch.nn.Module, traced : torch.fx.GraphModule,
                     node.replace_all_uses_with(get_attr_nodes[node.target])
                     traced.graph.erase_node(node)
 
+        for node in traced.graph.nodes:
+            if (node.op, node.target) == ('call_function', pipe_split):
+                nxt = node._next
```

Comment:
I would just split this into two loops: one to collect redundant pipe_split nodes in a list, then another that just does `erase_node` on all of them. I'd rather not have to reason about the validity of the pointers in the doubly-linked list while reading this

Reply:
yea i had that first and then saw that the just previous loop mutates (erases as well) during forward iteration so i figured it might be safe. so i went fishing in the impl of [`_node_list.__iter__`](https://github.com/pytorch/pytorch/blob/09be44de7b56495bcb5ad1d47376200cbb853097/torch/fx/graph.py#L233) and it looks like erased nodes are just tombstoned rather than actually deleted (i'm guessing that's why you have to call `traced.recompile()` after making changes). maybe @Chillee can weigh in here (as my goto on all things fx).

Reply:
Just because you can doesn't mean you should. I would reject this change as written purely from a readability standpoint

Reply:
even though I can merge I'd like your sign off :)